magma/syntax/sequential2.py

```diff
@@ -448,6 +413,42 @@ def seq_inner(cls):
 
         io_args = build_io_args(_annotations, output_port_names)
 
+        cls_setattr = cls.__setattr__
+        cls_getattribute = cls.__getattribute__
+
+        def sequential_getattribute(self, key):
+            """
+            Wrap lists so we can use the setattr interface with lists of registers
+            """
+            result = cls_getattribute(self, key)
```

Comment:
Would calling super() work here?

Reply:
No.  There is no inheritance going on.  If we did something like the follow could use super but its more of a change:
```Python
class new_cls(cls): 
     def __setattr__(self, attr):
            ... # body of seq_setattr
      
      def __getattribute__(self, attr, value): 
           ... # similar
```

Reply:
Ah yes, I see now, it's being "overridden" via monkey patch rather than subclassing.